server/src/main/java/org/opensearch/cluster/metadata/IndexAbstractionResolver.java

```diff
@@ -105,7 +106,7 @@ public List<String> resolveIndexAbstractions(
 
             // we always need to check for date math expressions
             final String dateMathName = indexNameExpressionResolver.resolveDateMathExpression(indexAbstraction);
-            if (dateMathName != indexAbstraction) {
+            if (!Objects.equals(dateMathName, indexAbstraction)) {
```

Comment:
I don't think either of the 2 is null - can you verify once. If they are non-null than `s1.equals(s2)` will be slightly (unnoticeable but still) faster.